component/src/main/java/org/wso2/extension/siddhi/io/cdc/source/CDCSource.java

```diff
@@ -0,0 +1,342 @@
+            if (isDirectoryCreated && log.isDebugEnabled()) {
+                log.debug("Directory created for history file.");
+            }
+        }
+
+        try {
+            Map<String, Object> configMap = CDCSourceUtil.getConfigMap(username, password, url, tableName,
+                    historyFileDirectory, siddhiAppName, streamName, serverID, serverName, connectorProperties,
+                    this.hashCode());
+            changeDataCapture.setConfig(configMap);
+        } catch (WrongConfigurationException ex) {
+            throw new SiddhiAppCreationException("The cdc source couldn't get started. Invalid" +
+                    " configuration parameters.", ex);
+        }
+    }
+
+    /**
+     * Returns the list of classes which this source can output.
+     *
+     * @return Array of classes that will be output by the source.
+     * Null or empty array if it can produce any type of class.
+     */
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[]{Map.class};
+    }
+
+    /**
+     * Initially Called to connect to the debezium embedded engine to receive change data events asynchronously.
+     */
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+        //keep the object reference in Object keeper
+        cdcSourceObjectKeeper.addCdcObject(this);
+
+        try {
+            executorService.execute(changeDataCapture.getEngine());
+        } catch (NullPointerException ex) {
+            throw new ConnectionUnavailableException("Connection is unavailable. Check parameters.", ex);
+        }
+    }
+
+    /**
+     * This method can be called when it is needed to disconnect from the end point.
+     */
+    @Override
+    public void disconnect() {
+    }
+
+    /**
+     * Called at the end to clean all the resources consumed by the {@link Source}
+     */
+    @Override
+    public void destroy() {
+        //Remove this CDCSource object from the CDCObjectKeeper.
+        cdcSourceObjectKeeper.removeObject(this.hashCode());
+
+        //shutdown the executor service.
+        executorService.shutdown();
+    }
+
+    /**
+     * Called to pause event consumption
+     */
+    @Override
+    public void pause() {
+        changeDataCapture.pause();
+    }
+
+    /**
+     * Called to resume event consumption
+     */
+    @Override
+    public void resume() {
+        changeDataCapture.resume();
+    }
+
+    /**
+     * Used to collect the serializable state of the processing element, that need to be
+     * persisted for the reconstructing the element to the same state on a different point of time
+     *
+     * @return stateful objects of the processing element as a map
+     */
+    @Override
+    public Map<String, Object> currentState() {
+        Map<String, Object> currentState = new HashMap<>();
+        currentState.put("cacheObj", offsetData);
```

Comment:
Shall we move "cacheObj" to a constant?

Reply:
Fixed with commit: `Fix some suggested issues in PR#1.` 8a89f22